### PR TITLE
[CMS-708] Convert composer command should also convert a build tools site.

### PR DESCRIPTION
**Pull request by kporras07**

### CODE REVIEW

**Review by namespacebrian — APPROVED** [review]
seems legit

File: src/Commands/ConvertToComposerSiteCommand.php [hunk 1]
```diff
@@ -126,18 +151,71 @@ public function convert(
...
+        }
+
         if (!$options['dry-run']) {
-            $this->pushTargetBranch();
-            $this->addCommitToTriggerBuild();
-            $this->executeDrushDatabaseUpdates($options);
-            $this->executeDrushCacheRebuild($options);
+            if ($treatAsBuildToolsSite) {
+                $this->pushTargetBranchBuildTools();
+                $this->log()->notice('Push done to external VCS repository.');
```
Comment (jabxjab) [inline src/Commands/ConvertToComposerSiteCommand.php:165]: @kporras07 How about to add `$this->log()->notice()` for `else` too?

File: src/Commands/ConvertToComposerSiteCommand.php [hunk 2]
```diff
@@ -335,16 +413,26 @@ private function copyPantheonYml(): void
 
         $path = Files::buildPath($this->getLocalSitePath(), 'pantheon.yml');
         $pantheonYmlContent = Yaml::parseFile($path);
-        if (isset($pantheonYmlContent['build_step']) && true === $pantheonYmlContent['build_step']) {
-            return;
+
+        $changed = false;
```
Comment (jabxjab) [inline src/Commands/ConvertToComposerSiteCommand.php:417]: @kporras07 There is a way not to introduce `$changed` var:
```php
$pantheonYmlContent = $pantheonYmlContentOriginal = Yaml::parseFile($path);

// ...

if (array_diff($pantheonYmlContent, $pantheonYmlContentOriginal)) {
    $pantheonYmlFile = fopen($path, 'wa+');
     fwrite($pantheonYmlFile, Yaml::dump($pantheonYmlContent, 2, 2));
     fclose($pantheonYmlFile);
}

File: src/Commands/Traits/ConversionCommandsTrait.php [hunk 3]
```diff
@@ -70,7 +71,11 @@ protected function cloneSiteGitRepository(bool $force = true): string
             sprintf('Cloning %s site repository into "%s"...', $this->site->getName(), $path)
         );
 
-        $this->getLocalMachineHelper()->cloneGitRepository($this->getRemoteGitUrl(), $path, true);
+        if (!$remoteGitUrl) {
+            $remoteGitUrl = $this->getRemoteGitUrl();
```
Comment (jabxjab) [inline src/Commands/Traits/ConversionCommandsTrait.php:75]: @kporras07 Tip (just a personal preference) - a ternary operator variant: 
```php
$remoteGitUrl = $remoteGitUrl ?: $this->getRemoteGitUrl();

File: src/Commands/Traits/ConversionCommandsTrait.php [hunk 4]
```diff
@@ -111,7 +116,7 @@ protected function getRemoteGitUrl(): string
      * @throws \Psr\Container\ContainerExceptionInterface
      * @throws \Pantheon\TerminusConversionTools\Exceptions\Git\GitException
      */
-    protected function getLocalSitePath(?bool $force = null)
+    protected function getLocalSitePath(?bool $force = null, string $remoteGitUrl = null)
```
Comment (jabxjab) [inline src/Commands/Traits/ConversionCommandsTrait.php:119]: @kporras07 Missing `@param`

File: src/Commands/Traits/ConversionCommandsTrait.php [hunk 5]
```diff
@@ -515,6 +547,20 @@ protected function isBuildToolsSite(): bool
         return in_array('build-metadata.json', $files);
     }
 
+    /**
+     * Get external vcs from build-metadata file.
+     *
+     * @throws \Pantheon\TerminusConversionTools\Exceptions\Git\GitException
+     */
+    protected function getExternalVcsUrl(): ?string
```
Comment (jabxjab) [inline src/Commands/Traits/ConversionCommandsTrait.php:555]: @kporras07 Missing `@return`

File: src/Commands/Traits/ConversionCommandsTrait.php [hunk 6]
```diff
@@ -515,6 +547,20 @@ protected function isBuildToolsSite(): bool
...
+    {
+        $buildMetadataFile = Files::buildPath($this->getLocalSitePath(), 'build-metadata.json');
+        $buildMetadataContent = json_decode(file_get_contents($buildMetadataFile), true);
+        if (isset($buildMetadataContent['url'])) {
```
Comment (jabxjab) [inline src/Commands/Traits/ConversionCommandsTrait.php:559]: @kporras07 
```php
return $buildMetadataContent['url'] ?? null;
```

**Review by jabxjab — CHANGES_REQUESTED** [review]
Some very minor changes (phpdoc updates)

**Comment by kporras07** [comment]
Feedback addressed, thanks

**Review by jabxjab — APPROVED** [review]

**Pull request merged**